chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomEventParserSpec.kt

```diff
@@ -0,0 +1,79 @@
+package com.pusher.chatkit.rooms
+
+import com.google.common.truth.Truth.assertThat
+import com.pusher.chatkit.TestFileReader
+import com.pusher.chatkit.rooms.api.CreateRoomResponse
+import com.pusher.chatkit.rooms.api.JoinRoomResponse
+import com.pusher.chatkit.rooms.api.JoinedRoomApiType
+import com.pusher.chatkit.util.parseAs
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object RoomEventParserSpec : Spek({
+    val testFileReader = TestFileReader("/json/rooms")
+
+    describe("when create room response from the docs is parsed") {
+
+        val result = testFileReader.readTestFile("room_created_response-docs.json")
+                .parseAs<CreateRoomResponse>().successOrThrow()
+
+        it("then the result has expected room") {
+            val expectedRoom = JoinedRoomApiType(
+                    "ac43dfef",
+                    "alice",
+                    "Chatkit chat",
+                    null,
+                    false,
+                    mapOf("highlight_color" to "blue"),
+                    "2020-01-08T14:55:10Z",
+                    "2017-03-23T11:36:42Z",
+                    "2017-03-23T11:36:42Z",
+                    null)
+
+            assertThat(result.room).isEqualTo(expectedRoom)
+        }
+
+        it("creates valid membership") {
+            val membership = result.membership
+
+            assertThat(membership.roomId).isEqualTo("ac43dfef")
+            assertThat(membership.userIds.size).isEqualTo(2)
+            assertThat(membership.userIds[0]).isEqualTo("alice")
+            assertThat(membership.userIds[1]).isEqualTo("carol")
+        }
+
+    }
+
+    describe("when joining a room example from the docs") {
+
+        val result = testFileReader.readTestFile("room_joined_response-docs.json")
+                .parseAs<JoinRoomResponse>().successOrThrow()
+
+        it("creates a valid room") {
+            val expectedRoom = JoinedRoomApiType(
```

Comment:
It's the same for joining as for creating a room. Indeed it's the same javascript in mimir. Could we have that shared as suggested previously [here](https://github.com/pusher/chatkit-android/pull/150#discussion_r370258283)? Indeed it would be good to share the json file as well: `create_or_join_room_response-docs.json`.

Reply:
Okay. I misunderstood I thought you were asking me to change the file name, not make them both point to the same json! Have updated to point to the same json file.